### PR TITLE
Add planned enhancements

### LINKED ISSUE #279: [PE-D][Tester B] Error message for view command is not specific enough
As shown in the screenshot, instead of "invalid command format", the error message for handling non-positive integers can be more specific. Eg: can provide the reason of the error like "Index is a non-positive integer!"

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: JiaYi-Gallium369/ped#3

### LINKED ISSUE #284: [PE-D][Tester B] Hard-to-type long command formats
Commands such as `scheduledate` and `scheduleall` may be too long and tedious to type, which may not fit the value proposition of  managing patients faster.

**Suggestion:** Support both a short version (easier to type) and a longer (easier to remember) version for a keyword (Example: Make `scheduledate` and `sdate` equivalent).

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: JiaYi-Gallium369/ped#4

### LINKED ISSUE #300: [PE-D][Tester A] Hard limit on appointment remarks character count seems too short for given use case in hospital
Given that the usecase is for patient management, the hard limit of character lengths on certain fields seem inappropriate

For instance, a doctor may need to write pre-surgery notes like this 
```
Patient brought into OR under stable anesthesia. Prepped and draped in the usual sterile fashion. Quick review of pre-op images showed severe blockages in the left anterior descending (LAD) and right coronary arteries (RCA). Decision confirmed for a double bypass using the left internal mammary artery (LIMA) for the LAD and a saphenous vein graft (SVG) for the RCA.
```
Which I thought was reasonable in length. But was rejected by the system

-------------
Labels: `type.FeatureFlaw` `severity.Medium`
original: AppleJem/ped#8

### LINKED ISSUE #337: [PE-D][Tester E] Inaccurate error message for view command
Steps:
Enter command: view 1123213123123

Expected behavior:
This was tested on a patient list with fewer patients than the index has specified. The error message should indicate that the patient is not valid.

Actual behavior:

The actual error message states that the command format is invalid, when in fact it is valid.

-------------
Labels: `severity.Medium` `type.FunctionalityBug`
original: zacharyxzz/ped#9

### PR DESCRIPTION
**Pull request by shuckycheese**
Added future planned enhancements to solve some current issues in WardWatch

Planned enhancements:
-more specific index out of bounds messages
-addition of appointment title and enhancement and viewappt command
-addition of command shortcuts

close #279 
close #284
close #300
close #337

### CODE REVIEW

**Review by niclammm — COMMENTED** [review]
lgtm

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -276,11 +276,32 @@ _{Explain here how the data archiving feature will be implemented}_
...
+In the case where the index passed in is negative, the error message `INDEX provided must be a positive integer, please try again!` will be shown
+<br><br>
+
+### Add appointment title and description and `viewappt` command
+Currently, our `makeappt` command only allows users to create appointments with a short description. This prevents users from adding too much information to the current description which may be restrictive.<br><br>
+**Planned implementation**<br><br>
+We plan to separate appointment title and description which gives users more flexibility. Appointment title will just be a short description such as `Surgery` or `Medical checkup` and support a character limit of 40. Appointment description will be for additional information about the appointment that the user may want to add. The appointment description will support a much longer character limit of 300 but will now be shown in the appointment list panel as the long inputs may cause issues with the UI. <br>
```
Comment (choonzies) [inline docs/DeveloperGuide.md:295]: I would like to check it the "now" a typo and supposed to be "not"?

**Review by choonzies — COMMENTED** [review]
Potential typo, the rest looks good!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (shuckycheese) [reply docs/DeveloperGuide.md]: Good catch! I've made the necessary amends.

**Pull request merged**